qiskit/transpiler/passes/scheduling/base_scheduler.py

```diff
@@ -0,0 +1,272 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2020.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Base circuit scheduling pass."""
+
+from typing import Dict
+from qiskit.transpiler import InstructionDurations
+from qiskit.transpiler.basepasses import TransformationPass
+from qiskit.transpiler.passes.scheduling.time_unit_conversion import TimeUnitConversion
+from qiskit.dagcircuit import DAGOpNode, DAGCircuit
+from qiskit.circuit import Delay, Gate
+from qiskit.circuit.parameterexpression import ParameterExpression
+from qiskit.transpiler.exceptions import TranspilerError
+
+
+class BaseScheduler(TransformationPass):
+    """Base scheduler pass.
+
+    Policy of topological node ordering in scheduling
+
+        The DAG representation of ``QuantumCircuit`` respects the node ordering also in the
+        classical register wires, though theoretically two conditional instructions
+        conditioned on the same register are commute, i.e. read-access to the
+        classical register doesn't change its state.
+
+        .. parsed-literal::
+
+                 ┌────────────────┐   ┌───┐
+            q_0: ┤ Delay(100[dt]) ├───┤ X ├───
+                 └─────┬───┬──────┘   └─╥─┘
+            q_1: ──────┤ X ├────────────╫─────
+                       └─╥─┘            ║
+                    ┌────╨────┐    ┌────╨────┐
+            c: 1/═══╡ c_0=0x1 ╞════╡ c_0=0x1 ╞
+                    └─────────┘    └─────────┘
```

Comment:
```suggestion
            qc = QuantumCircuit(2, 1)
            qc.delay(100, 0)
            qc.x(0).c_if(0, True)
            qc.x(1).c_if(0, True)
```
Thank you for the quick addition of this chapter, it's very informative.
Unfortunately, due to the bug in text-drawer, we might have to show the original circuit representation like above here...